packages/api/src/controllers/stream.ts

```diff
@@ -1170,12 +1170,22 @@ app.post("/:id/lockPull", authorizer({ anyAdmin: true }), async (req, res) => {
     return res.json({ errors: ["not found"] });
   }
 
+  const doingActiveCleanup = activeCleanupOne(
+    req.config,
+    stream,
+    req.queue,
+    await getIngestBase(req)
+  );
+  // the `isActive` field is only cleared later in background, so we ignore it
+  // in the query below in case we triggered an active cleanup logic above.
+
+  const leaseDeadline = Date.now() - leaseTimeout;
   const updateRes = await db.stream.update(
     [
       sql`id = ${stream.id}`,
-      sql`(data->>'pullLockedBy' = ${host} OR (COALESCE((data->>'pullLockedAt')::bigint,0) < ${
-        Date.now() - leaseTimeout
-      } AND COALESCE((data->>'isActive')::boolean,FALSE) = FALSE))`,
+      doingActiveCleanup
+        ? sql`(data->>'pullLockedBy' = ${host} OR (COALESCE((data->>'pullLockedAt')::bigint,0) < ${leaseDeadline}))`
+        : sql`(data->>'pullLockedBy' = ${host} OR (COALESCE((data->>'pullLockedAt')::bigint,0) < ${leaseDeadline} AND COALESCE((data->>'isActive')::boolean,FALSE) = FALSE))`,
```

Comment:
@leszko I could also replicate the `lastSeen` logic here but this query would become even more complex, so I preferred to keep it "non-concurrent-safe" for now. Let me know if you think we need to go all the way here.

Reply:
Yeah, I think it's fine as it is.